exercises/practice/grains/.docs/instructions.append.md

```diff
@@ -0,0 +1,12 @@
+## Maximum integer value
+
+According to [the Vim docs][number]:
+
+> Assuming 64 bit numbers are used (see v:numbersize) an unsigned number is truncated to 0x7fffffffffffffff or 9223372036854775807.
+
+That means that Vim cannot express any number `2^63` or greater as an integer.
+
+For this exercise, you will return the values as strings.
+That means you will have to implement a way to do addition on two _string_ operands.
```

Comment:
I am not crazy about this line.  Addition is not the only way to do this, so we will not have to necessarily do this, but it states that we will have to  do so.  The word "implement", with its definitions, is ambiguous, and it may the solution does not use (at least only) two operands of any kind.

Also, the exercise will have us return the values as strings, but that does not mean what the line after says it means.

Perhaps part of my angst here is the "that means" part that is stated 4 lines above as well.

How can we phrase this so that the restrictions in Vimscript is stated, but the solution is left to the exerciser to be thought and explored.

Reply:
```none
In other words, Vimscript cannot express any number `2^63` or greater as an integer.

Some of the tests for this exercise require 64 bit integers and thus cannot be solved using only integer arithmetic.
The tests have been written to expect the return values as strings.
How can you implement arithmetic operations that return strings?
```

Reply:
> How can you implement arithmetic operations that return strings?

Maybe:

"Can you solve this by avoiding numbers that are larger than the language will allow directly?"

The tests will inform that the returns must be strings, and if the tests change later, then we do not have to change that aspect.  The communication is (hopefully) clear.

> and thus cannot be solved using only integer arithmetic.

"which is beyond the integer size limitation of Vimscript."

> The tests have been written to expect the return values as strings.

"Because of the language limitations regarding integers, the results of the calculations are tested against a string which expresses the integer value, rather than expressing the answer as Integer."

Reply:
Also, notice that I do avoid the use of "implement" as the word has confusing definitions.  "The implement implements an implementation, within another implement."  This statement is clear, but unless you are a native English speaker, it is may be hard to follow.